app/angular/src/client/preview/angular-beta/utils/NgComponentAnalyzer.ts

```diff
@@ -136,5 +136,14 @@ export const getComponentDecoratorMetadata = (component: any): Component | undef
       ? Reflect.getOwnPropertyDescriptor(component, decoratorKey).value
       : component[decoratorKey];
 
-  return (decorators || []).find((d) => d instanceof Component);
+  if (!decorators) {
```

Comment:
Could you add some test on this file ? https://github.com/storybookjs/storybook/blob/5bcfdae1b6f5d7849a0a48b58a306410977e0cf5/app/angular/src/client/preview/angular-beta/utils/NgComponentAnalyzer.test.ts

To run simple test `yarn jest NgComponentAnalyzer.test.ts` or with watch mode `yarn jest NgComponentAnalyzer.test.ts --watch`
don't hesitate to ask me for help if needed ;)

Reply:
Hey, I tried my best to create a test at least for the "regular" case. That is, when you are trying to use a component which is not from an Angular library bundle. That test didn't exist indeed. However, once you compile the components into a library, everything becomes very "plain" and we cannot take advantage of types that much to create clean tests, or at least I could not come up with a way to do that in a clean way. So what here is some evidence, if you will, just for the sake of traceability in regards to the expected behavior:

![image](https://user-images.githubusercontent.com/35342506/130243741-a0220277-c9f8-442c-a177-4874df7f9aab.png)

Instead of:
![image](https://user-images.githubusercontent.com/35342506/130244743-29774c0c-96bc-472e-93da-3ac7ac68a48f.png)

Reply:
👍 
Top these evidence if you want you can add them here :

https://github.com/storybookjs/storybook/tree/1508807079ecb35a34f8f05c7d60a179b770504a/examples/angular-cli/src/stories/basics
As you can see there are others.  It's also a way to test the application when unit testing is not efficient / possible 
The examples try to be autonomous and represent a basic use case

Reply:
I'm working on a minimal Angular Library setup for that `angular-cli` workspace. I'll let you know when I'm done and we can discuss some minor details. But we'll cross that bridge when we get to it.